enterprise/server/ociregistry/ociregistry.go

```diff
@@ -0,0 +1,305 @@
+package ociregistry
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"net/http"
+
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/environment"
+	"github.com/buildbuddy-io/buildbuddy/server/metrics"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/prefix"
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+)
+
+const (
+	rangeHeaderBytesPrefix = "bytes="
+
+	registryInstanceName = "registry_artifacts"
```

Comment:
unused?

Reply:
but also, if you're planning to use this the way I think you are, wdyt of making it a flag so we can nuke the registry cache by changing the flag value in the config if needed?

Reply:
👍 Taking this out until we need it, but when the time comes, yes, let's make it a flag.